core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -538,8 +557,20 @@ public long nextFetchOffset() {
             }
 
             long nextFetchOffset = -1;
-
+            long gapStartOffset = initialReadGapOffset != null ? initialReadGapOffset.gapStartOffset() : -1;
```

Comment:
nit: but better to be consistent.
```suggestion
            long gapStartOffset = isInitialReadGapOffsetWindowActive() ? initialReadGapOffset.gapStartOffset() : -1;
```